packages/compiler/src/core/checker.ts

```diff
@@ -2090,6 +2090,21 @@ export function createChecker(program: Program): Checker {
         return undefined;
       }
 
+      if (base.flags & SymbolFlags.TemplateParameter) {
+        const constraint = (base.declarations[0] as TemplateParameterDeclarationNode).constraint;
+        if (
+          constraint &&
+          (constraint.kind === SyntaxKind.MemberExpression ||
+            constraint.kind === SyntaxKind.TypeReference ||
+            constraint.kind === SyntaxKind.Identifier)
+        ) {
+          base = resolveTypeReferenceSym(constraint, mapper, resolveDecorator);
```

Comment:
Right now when we resolve a template declaration type (uninstantiated) and you reference a template parameter with constrain the reference remains the template parameter but with this it would become the type in the constraint.

```tsp
alias Options = {name: string};

model Foo<T extends Options> {
  n: T.name; // This resolve to `Scalar: string`
  t: T;            // This resolve to `TemplateParameter: T`
}
```

Wondering if then we need a new Type `TemplateParameterMember` or something like that to represent that accurately and not loose information